src/tables.jl

```diff
@@ -0,0 +1,87 @@
+# Tables and TableTraits interface
+
+## Chains
+
+Tables.istable(::Type{<:Chains}) = true
+
+Tables.columnaccess(::Type{<:Chains}) = true
+
+Tables.columns(chn::Chains) = chn
+
+Tables.columnnames(chn::Chains) = (:Iteration, :Chain, Symbol.(names(chn))...)
```

Comment:
I'm wondering if we should use `:iteration` and `:chain`?

Reply:
Also, do we have to consider the (not very likely) possibility that one of the names clashes with `:iteration` or `:chain`?

Reply:
Yeah, I was debating this. The summary dataframe columns use the lower case names, so there's a point in that favor. I ultimately went with upper case since it seemed less likely those would collide with a parameter name in a model. I'm happy to change it to lower case if you like.

Reply:
Oh, I see I was writing my reply as you were writing my comment. There are a few options here: 1) choose a column name that is highly unlikely a user would use (e.g. `:__chain:`) 2) choose a column name based on the variable names (e.g. if a user has `:chain` then use `:chainid`) 3) throw an error if `:chain` is in the parameter names 4) do nothing and hope the user doesn't use one of those names.

I really don't like 1) and 2). 3) is fine. I do think it's unlikely a user would name a parameter `Chain` or `Iteration` if following Julia conventions, but it is certainly possible.

Reply:
Alternatively one could prefix all parameter names with something like `param_` - but IMO that's pretty bad from a user perspective even though it would solve the issue :stuck_out_tongue:

Reply:
> Alternatively one could prefix all parameter names with something like `param_` - but IMO that's pretty bad from a user perspective even though it would solve the issue 😛

I agree. I'm not certain what the best approach here is and don't have a preference. I'm warming up to 1), but I'll defer to others with more of a preference.

Reply:
Actually, I was leaning towards 3), since if a user used `:chain` or `:iteration` as parameter name, she could just change it with `set_names` or in the upstream code (and I strongly believe there shouldn't be any "magic" renaming in the case of name clashes). But how could we reliably throw an error in 3)? A user (or developer) doesn't have (?) to call `columnnames`, but `getcolumn` would still be problematic. So if we want to throw an error, at which stage should this be done?

Reply:
I think the right way is to perform the check in `Tables.rows` and `Tables.columns`, as well as `Tables.schema` to be safe, since these are the [core interface functions](https://juliadata.github.io/Tables.jl/stable/#Implementing-the-Interface-(i.e.-becoming-a-Tables.jl-source)-1) and are recommended to be called before calling functions like `getcolumn`. I'll go ahead and implement that as well as name changes to `:chain` and `:iteration` tomorrow.

Reply:
Sounds good, thanks!

Reply:
Ended up implementing it tonight.